nuklear_gamepad.h

```diff
@@ -84,14 +99,15 @@ struct nk_gamepads {
     struct nk_gamepad gamepads[NK_GAMEPAD_MAX];
     struct nk_context* ctx;
     void* user_data;
```

Comment:
With `input_source` having its own `user_data`, do we still need `nk_gamepad::user_data`? I guess it doesn't hurt to have it.

Reply:
Good point, yeah I don't think it's needed anymore. I left a user_data param on nk_gamepad_init to pass to the default input source that gets created (the pntr backend needs this)